presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcDiagnosticModule.java

```diff
@@ -28,35 +30,28 @@
 
 import static com.google.common.reflect.Reflection.newProxy;
 import static java.lang.String.format;
-import static java.util.Objects.requireNonNull;
 import static org.weakref.jmx.guice.ExportBinder.newExporter;
 
 public class JdbcDiagnosticModule
         implements Module
 {
-    private final String catalogName;
-
-    public JdbcDiagnosticModule(String catalogName)
-    {
-        this.catalogName = requireNonNull(catalogName, "catalogName is null");
-    }
-
     @Override
     public void configure(Binder binder)
     {
         binder.install(new MBeanServerModule());
         binder.install(new MBeanModule());
 
+        Provider<CatalogName> catalogName = binder.getProvider(CatalogName.class);
```

Comment:
Squash with `Inject CatalogName in JDBC connectors`?

the `Inject CatalogName in JDBC connectors` commit looks so void on its own